### DISCUSSION TITLE
@tsed/socketio-testing - test with two different clients?

### DISCUSSION THREAD

**Discussion by Ravendarksky** [TypedProject/ts-express-decorators#2324] [Q&A] [answered]
Good evening all,

I am reading through https://tsed.io/tutorials/socket-io.html#testing and trying to understand how I can write a test that simulates two clients connecting to the same socket.io server?  It is not clear to me. 

In the example and from looking at the source it seems like 

`await service.get`

will always return the same client.

What am I missing or is this library simply not useful for my purpose?

I was hoping to do something like this:

```
const service = TestMongooseContext.get<SocketClientService>(SocketClientService);

const client = await service.get('/someendpoint');

const client2 = await service.get('/someendpoint');

expect(client.id).not.toEqual(client2.id);
```

**Comment by Romakita** [comment, marked as answer]
Hello @Ravendarksky 

Here is the explanation:

```ts
TestMongooseContext.get<SocketClientService>(SocketClientService) => // singleton
await TestMongooseContext.invoke<SocketClientService>(SocketClientService)  => // new instance
```

See you
Romain
